packages/utils/src/lib/scoring.ts

```diff
@@ -42,31 +42,45 @@ function categoryRefToScore(
   audits: EnrichedAuditReport[],
   groups: EnrichedScoredAuditGroup[],
 ): (ref: CategoryRef) => number {
+  // Create lookup objects
+  const auditLookup = audits.reduce<Record<string, EnrichedAuditReport>>(
+    (lookup, audit) => {
+      lookup[`${audit.plugin}-${audit.slug}`] = audit;
+      return lookup;
+    },
+    {},
+  );
+
+  const groupLookup = groups.reduce<Record<string, EnrichedScoredAuditGroup>>(
+    (lookup, group) => {
+      lookup[`${group.plugin}-${group.slug}`] = group;
+      return lookup;
+    },
+    {},
+  );
```

Comment:
Same as before:
- should probably be `reduce` and no mutation, or
- if it has a performance impact, keep the mutation, but use `for` or `.forEach()`

Reply:
I bed mutating in for each is faster. lets see the benchmarks

Reply:
I'll keep the declarative one, but will check both in benchmark